src/Properties/SchemaAggregator.php

```diff
@@ -0,0 +1,398 @@
+                && $stmt->name->name === 'up'
+                && $stmt->stmts
+            ) {
+                $this->addUpMethodStatements($stmt->stmts);
+            }
+        }
+    }
+
+    /**
+     * @param PhpParser\Node\Stmt[] $stmts
+     */
+    private function addUpMethodStatements(array $stmts): void
+    {
+        foreach ($stmts as $stmt) {
+            if ($stmt instanceof PhpParser\Node\Stmt\Expression
+                && $stmt->expr instanceof PhpParser\Node\Expr\StaticCall
+                && ($stmt->expr->class instanceof PhpParser\Node\Name)
+                && $stmt->expr->name instanceof PhpParser\Node\Identifier
+                && $stmt->expr->class->toCodeString() === '\\Illuminate\Support\Facades\Schema'
+            ) {
+                switch ($stmt->expr->name->name) {
+                    case 'create':
+                        $this->alterTable($stmt->expr, true);
+                        break;
+
+                    case 'table':
+                        $this->alterTable($stmt->expr, false);
+                        break;
+
+                    case 'drop':
+                    case 'dropIfExists':
+                        $this->dropTable($stmt->expr);
+                        break;
+
+                    case 'rename':
+                        $this->renameTable($stmt->expr);
+                }
+            }
+        }
+    }
+
+    private function alterTable(PhpParser\Node\Expr\StaticCall $call, bool $creating): void
+    {
+        if (! isset($call->args[0])
+            || ! $call->args[0]->value instanceof PhpParser\Node\Scalar\String_
+        ) {
+            return;
+        }
+
+        $tableName = $call->args[0]->value->value;
+
+        if ($creating) {
+            $this->tables[$tableName] = new SchemaTable($tableName);
+        }
+
+        if (! isset($call->args[1])
+            || ! $call->args[1]->value instanceof PhpParser\Node\Expr\Closure
+            || count($call->args[1]->value->params) < 1
+            || ($call->args[1]->value->params[0]->type instanceof PhpParser\Node\Name
+                && $call->args[1]->value->params[0]->type->toCodeString()
+                !== '\\Illuminate\Database\Schema\Blueprint')
+        ) {
+            return;
+        }
+
+        $updateClosure = $call->args[1]->value;
+
+        if ($call->args[1]->value->params[0]->var instanceof PhpParser\Node\Expr\Variable
+            && is_string($call->args[1]->value->params[0]->var->name)
+        ) {
+            $argName = $call->args[1]->value->params[0]->var->name;
+
+            $this->processColumnUpdates($tableName, $argName, $updateClosure->stmts);
+        }
+    }
+
+    /**
+     * @param string                $tableName
+     * @param string                $argName
+     * @param PhpParser\Node\Stmt[] $stmts
+     *
+     * @throws \Exception
+     */
+    private function processColumnUpdates(string $tableName, string $argName, array $stmts): void
```

Comment:
It's ok to maintain all of this?

Reply:
I think yes. I tested the code on many different Laravel projects. It is working well. Also, psalm Laravel plugin uses this. So if there is any issue we can help each other.